src/utils/window-console.js

```diff
@@ -148,6 +149,54 @@ export function addDataToWindowObject(
     `);
   };
 
+  target.retrieveProfileFromBrowser = async function (): Promise<
+    MixedObject | ArrayBuffer | null,
+  > {
+    // Note that a new connection is created instead of reusing the one in the
+    // redux state, as an attempt to make it work even in the worst situations.
+    const browserConnectionStatus = await createBrowserConnection();
+    const browserConnection = actions.unwrapBrowserConnection(
+      browserConnectionStatus
+    );
+    const rawGeckoProfile = await browserConnection.getProfile({
+      onThirtySecondTimeout: () => {
+        console.log(
+          oneLine`
+            We were unable to connect to the browser within thirty seconds.
+            This might be because the profile is big or your machine is slower than usual.
+            Still waiting...
+          `
+        );
+      },
+    });
+
+    return rawGeckoProfile;
+  };
+
+  target.saveToDisk = async function (
+    rawGeckoProfile: MixedObject | ArrayBuffer,
```

Comment:
I guess this could any object as well like you said in Matrix and not only `rawGeckoProfile`. Maybe rename this to reflect that?